signer/src/network.rs

```diff
@@ -14,89 +14,74 @@ use crate::message;
 
 pub type Msg = ecdsa::Signed<message::SignerMessage>;
 
+/// Represents the interaction point between signers and the signer network,
+/// allowing signers to exchange messages with each other.
 pub trait MessageTransfer {
-    type Error;
+    type Error: std::error::Error;
     /// Send `msg` to all other signers
     fn broadcast(&mut self, msg: Msg) -> impl Future<Output = Result<(), Self::Error>> + Send;
     /// Receive a message from the network
     fn receive(&mut self) -> impl Future<Output = Result<Msg, Self::Error>> + Send;
 }
 
 #[cfg(test)]
-mod tests {
-    use super::*;
-
-    use bitcoin::hashes::Hash;
-    use p256k1::scalar;
+pub async fn assert_clients_should_be_able_to_exchange_messages<
```

Comment:
What are your thoughts on this?
```rust
pub async fn assert_clients_can_exchange_messages<
```

Reply:
Yeah that's better. Thank you for the suggestion. I'll update it.

Reply:
Done.